akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -335,6 +334,154 @@ class ClusterSharding(system: ExtendedActorSystem) extends Extension {
     start(typeName, entityProps, settings, messageExtractor, allocationStrategy, PoisonPill)
   }
 
+  /**
+   * Scala API: Register a named entity type by defining a factory for the [[akka.actor.Props]] of
+   * the entity actor and functions to extract entity and shard identifier from messages. The
+   * [[ShardRegion]] actor for this type can later be retrieved with the [[#shardRegion]] method.
+   *
+   * Some settings can be configured as described in the `akka.cluster.sharding` section
+   * of the `reference.conf`.
+   *
+   * @param typeName the name of the entity type
+   * @param propsFactory function that, given an entity id, returns the `Props` of the entity actors
+   *  that will be created by the `ShardRegion`
+   * @param settings configuration settings, see [[ClusterShardingSettings]]
+   * @param extractEntityId partial function to extract the entity id and the message to send to the
+   *   entity from the incoming message, if the partial function does not match the message will
+   *   be `unhandled`, i.e. posted as `Unhandled` messages on the event stream
+   * @param extractShardId function to determine the shard id for an incoming message, only messages
+   *   that passed the `extractEntityId` will be used
+   * @param allocationStrategy possibility to use a custom shard allocation and
+   *   rebalancing logic
+   * @param handOffStopMessage the message that will be sent to entities when they are to be stopped
+   *   for a rebalance or graceful shutdown of a `ShardRegion`, e.g. `PoisonPill`.
+   * @return the actor ref of the [[ShardRegion]] that is to be responsible for the shard
+   */
+  def start(
+    typeName:           String,
+    propsFactory:       String ⇒ Props,
```

Comment:
The name `propsFactory` is a bit vague. Props of what? I think it should still be `entityProps`, or `entityPropsFactory`